modules/persistent-storage-csi-smb-cifs-provision-static.adoc

```diff
@@ -74,15 +74,15 @@ spec:
     - file_mode=0777
   csi:
     driver: smb.csi.k8s.io
-    volumeHandle: smb-server.default.svc.cluster.local/share/ <2>
+    volumeHandle: smb-server.default.svc.cluster.local/share#subdir#app-data <2>
     volumeAttributes:
       source: //<hostname>/<shares> <3>
     nodeStageSecretRef:
       name: <secret_name_shares> <4>
       namespace: <namespace> <5>
 ----
 <1> The name of the PV.
-<2> `volumeHandle` format: {smb-server-address}.{sub-dir-name}.{share-name}. Ensure that this value is unique for every share in the cluster.
+<2> `volumeHandle` format: {smb-server-address}#{sub-dir-name}#{share-name}. Ensure that this value is unique for every share in the cluster. The `sub-dir-name` is optional. If you are using a sub directory, the sub directory must be exist or the mount will fail.
```

Comment:
I cannot reproduce "mount fail" if subdir doesn't exist. Please share PV's yaml that worked for you this way.

Reply:
Hi @mpatlasov , the example is not good enough, if using the subPath the `csi.volumeAttributes` should also contains the `subdir`.
```console
oc apply -f - <<EOF
apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    pv.kubernetes.io/provisioned-by: smb.csi.k8s.io
  name: my-smb-pv 
spec:
  capacity:
    storage: 10Gi
  accessModes:
    - ReadWriteMany
  persistentVolumeReclaimPolicy: Retain
  storageClassName: ""
  mountOptions:
    - dir_mode=0777
    - file_mode=0777
  csi:
    driver: smb.csi.k8s.io
    volumeHandle: samba-server.samba-server.svc.cluster.local/share#sub-new#my-smb-pv
    volumeAttributes:
      source: //samba-server.samba-server.svc.cluster.local/share
      subdir: sub-new
    nodeStageSecretRef:
      name: smbcreds 
      namespace: samba-server
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pvc-static 
spec:
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 10Gi 
  storageClassName: ""
  volumeName: my-smb-pv
---
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  containers:
    - name: container
      image: quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339
      volumeMounts:
        - mountPath: /mnt/storage
          name: data
  volumes:
    - name: data
      persistentVolumeClaim:
        claimName: pvc-static
EOF

$ oc describe po mypod
...
Events:
  Type     Reason            Age               From               Message
  ----     ------            ----              ----               -------
  Warning  FailedScheduling  19s               default-scheduler  0/6 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/6 nodes are available: 6 Preemption is not helpful for scheduling.
  Normal   Scheduled         12s               default-scheduler  Successfully assigned default/mypod to ip-10-0-27-221.us-east-2.compute.internal
  Warning  FailedMount       4s (x5 over 12s)  kubelet            MountVolume.MountDevice failed for volume "my-smb-pv" : rpc error: code = Internal desc = volume(samba-server.samba-server.svc.cluster.local/share#sub-new#app-data) mount "//samba-server.samba-server.svc.cluster.local/share/sub-new" on "/var/lib/kubelet/plugins/kubernetes.io/csi/smb.csi.k8s.io/a1863686326ffd16b5de7342ec0206828eac57f43537e07c3ee69f4fc73ce307/globalmount" failed with mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t cifs -o dir_mode=0777,file_mode=0777,<masked> //samba-server.samba-server.svc.cluster.local/share/sub-new /var/lib/kubelet/plugins/kubernetes.io/csi/smb.csi.k8s.io/a1863686326ffd16b5de7342ec0206828eac57f43537e07c3ee69f4fc73ce307/globalmount
Output: mount error(2): No such file or directory
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs) and kernel log messages (dmesg)

```

Reply:
Oh, you used `subdir` in `volumeAttributes` -- this makes difference! I think both examples would be valid and reasonable: with `subdir` (as in your yaml above) and without it (as we currently have in our docs). My point is that _if_ we follow example without "subdir" in "volumeAttributes", _then_ it makes no sense to mention or explain the `subdir` in `volumeHandle` fomat.

Reply:
Agree.